azure/posix-sdist.yml

```diff
@@ -0,0 +1,128 @@
+        displayName: Create build venv with build dependices and build source dist
+
+      - bash: |
+          set -e
+
+          source build_env/bin/activate
+          python -m pip install twine
+
+          twine check scikit-learn/dist/*
+        displayName: Twine check
+      - bash: |
+          set -e
+
+          python -m venv $TEST_VENV
+          source $TEST_VENV/bin/activate
+          python --version
+          python -m pip install -U pip
+
+          # pyproject.toml will be released with >=0.23
+          if [[ "$BUILD_COMMIT" =~ ^0.22* ]]; then
+            python -m pip install Cython
+          fi
+
+          # Uses pep 517
+          pip install scikit-learn/dist/*.tar.gz
+        displayName: Create test venv and install using source dist
+
+      - bash: |
+          set -e
+          source $TEST_VENV/bin/activate
+
+          mkdir $TEST_DIR
+          cd $TEST_DIR
+
+          python -m pip install pytest
+          pytest -l --junitxml=$JUNITXML --pyargs sklearn
+        displayName: Runs tests
+
+      - task: PublishTestResults@2
+        inputs:
+          testResultsFiles: "$(TEST_DIR)/$(JUNITXML)"
+          testRunTitle: ${{ format('{0}-$(Agent.JobName)', 'Linux_Source_Dist') }}
+        displayName: "Publish Test Results"
+        condition: eq(variables['SKIP_BUILD'], 'false')
+
+      - bash: |
+          echo "##vso[task.prependpath]$CONDA/bin"
+          sudo chown -R $USER $CONDA
+        displayName: Add conda to PATH
+        condition: and(succeeded(), eq(variables['SKIP_BUILD'], 'false'), ne(variables['Build.Reason'], 'PullRequest'))
+
+      - bash: conda install -q -y anaconda-client
+        displayName: Install anaconda-client
+        condition: and(succeeded(), eq(variables['SKIP_BUILD'], 'false'), ne(variables['Build.Reason'], 'PullRequest'))
+
+      - bash: |
+          set -e
+          if [ "$BUILD_REASON" == "Schedule" ]; then
+            ANACONDA_ORG="scipy-wheels-nightly"
+            TOKEN="$SCIKIT_LEARN_NIGHTLY_UPLOAD_TOKEN"
+          else
+            ANACONDA_ORG="scikit-learn-wheels-staging"
+            TOKEN="$SCIKIT_LEARN_STAGING_UPLOAD_TOKEN"
+          fi
+          if [ "$TOKEN" == "" ]; then
+            echo "##[warning] Could not find anaconda.org upload token in secret variables"
+          fi
+          echo "##vso[task.setvariable variable=TOKEN]$TOKEN"
+          echo "##vso[task.setvariable variable=ANACONDA_ORG]$ANACONDA_ORG"
+        displayName: Retrieve secret upload token
+        condition: and(succeeded(), eq(variables['SKIP_BUILD'], 'false'), ne(variables['Build.Reason'], 'PullRequest'))
+        env:
+          # Secret variables need to mapped to env variables explicitly:
+          SCIKIT_LEARN_NIGHTLY_UPLOAD_TOKEN: $(SCIKIT_LEARN_NIGHTLY_UPLOAD_TOKEN)
+          SCIKIT_LEARN_STAGING_UPLOAD_TOKEN: $(SCIKIT_LEARN_STAGING_UPLOAD_TOKEN)
+
+      - bash: |
+          set -e
+          # The --force option forces a replacement if the remote file already
```

Comment:
could we maybe not attach a date or build number or something to the file name so that we don't override the existing files?

Reply:
I think this is done in the other configurations to be nice to the provider and not load tons of files onto their server. (In this case the provider is anaconda).

Is this correct @ogrisel ?

Reply:
But I don't think they really mind. As long as we remove things which are like a year old, then there shouldn't be an issue

Reply:
The `--force` is used in the other build scripts. Can we merge this with `--force` as this is the current behavior and have another pr on removing the `--force`?